### PR TITLE
Correct BaseOpenmrsObject equals

### PR DESCRIPTION
**Pull request by ShekharReddy4**
Trunk-3844-Correct BaseOpenmrsObject equals

### CODE REVIEW

**Comment by dkayiwa** [comment]
The formatting does not look correct. Did you run mvn clean install before committing? Can you do so now? Then make a followup commit which will have formatting changes. Finally squash the commits into one.

**Comment by dkayiwa** [comment]
Can you change the capitalization of the verifies text such that it is the same as in Rafal's original pull request?

**Comment by ShekharReddy4** [comment]
@dkayiwa  changed the capitalization of the verifies text same as in the rafal's original pull request, could you please review this.

File: api/src/test/java/org/openmrs/BaseOpenmrsObjectTest.java [hunk 1]
```diff
@@ -185,4 +188,130 @@ public void toString_shouldIncludeHashCodeIfUuidIsNull() throws Exception {
...
+	 */
+	@Test
+	public void equals_shouldReturnFalseIfGivenObjectIsNotSubclassOfSuperOrExtendingClass() throws Exception {
+		Order order = new Order(21);
+		OrderFrequency type = new OrderFrequency();
+		type.setUuid(order.getUuid());
+		Assert.assertFalse(type.equals(order));
+		Assert.assertFalse(order.equals(type));
+	}
+	/**
```
Comment (djazayeri) [inline api/src/test/java/org/openmrs/BaseOpenmrsObjectTest.java]: missing newline

**Comment by ShekharReddy4** [comment]
@dkayiwa  hi, https://issues.openmrs.org/browse/TRUNK-3844?focusedCommentId=193356&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-193356    in the above comment , it is mentioned that objects with same uuid are resulting in wrong behaviour of equals method. :)

**Comment by ShekharReddy4** [comment]
@dkayiwa  Please let me know in detail about what is the isuue with the Uuids used in my PR. :)

**Comment by dkayiwa** [comment]
Can you respond separately to each of my comments. Do you see the add a
line note button?

On Wed, Jan 27, 2016 at 12:32 AM, Shekhar Reddy <notifications@github.com>
wrote:

> @dkayiwa <https://github.com/dkayiwa> Please let me know in detail about
> what is the isuue with the Uuids used in my PR. :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openmrs/openmrs-core/pull/1655#issuecomment-175240711>
> .
>



-- 
If we keep uppermost in our minds the unkind and unjust acts of others, we
shall find it impossible to love them as Christ has loved us; but if our
thoughts dwell upon the wondrous love and pity of Christ for us, the same
spirit will flow out to others.

**Comment by dkayiwa** [comment]
@ShekharReddy4 can you squash these commits into one?

**Comment by ShekharReddy4** [comment]
@dkayiwa  squashed the commits :)

**Comment by ShekharReddy4** [comment]
@dkayiwa  thank u for your patience and help..A good community around me :)) Sorry for all my faults.

**Comment by dkayiwa** [comment]
@ShekharReddy4 i would not call them faults. Just call it learning process! :)
Thanks again for your great efforts and patience! :+1:

**Pull request merged**